scalding-hadoop-test/src/test/scala/com/twitter/scalding/ordered_serialization/OrderedSerializationTest.scala

```diff
@@ -0,0 +1,58 @@
+package com.twitter.scalding.ordered_serialization
+
+import com.twitter.scalding._
+import com.twitter.scalding.platform.{ HadoopPlatformJobTest, HadoopPlatformTest }
+import com.twitter.scalding.serialization.OrderedSerialization
+
+import org.scalacheck.{ Arbitrary, Gen }
+import org.scalatest.FunSuite
+
+import scala.language.experimental.macros
+import scala.math.Ordering
+
+object OrderedSerializationTest {
+  implicit val genASGK = Arbitrary {
+    for {
+      ts <- Arbitrary.arbitrary[Long]
+      b <- Gen.nonEmptyListOf(Gen.alphaNumChar).map (_.mkString)
+    } yield NestedCaseClass(RichDate(ts), (b, b))
+  }
+
+  def sample[T: Arbitrary]: T = Arbitrary.arbitrary[T].sample.get
+  val data = sample[List[NestedCaseClass]].take(1000)
+}
+
+case class NestedCaseClass(day: RichDate, key: (String, String))
+
+class OrderedSerializationTest extends FunSuite with HadoopPlatformTest {
+  import OrderedSerializationTest._
+  test("A test job with a fork and join, had previously not had boxed serializations on all branches") {
```

Comment:
this is only testing for no exception, right? Can we add a comment so we don't lose track of that later (or add also a logic test of what the results should be)?